tuna/miopen/driver/batchnorm.py

```diff
@@ -42,49 +44,52 @@
 class DriverBatchNorm(DriverBase):
   """Represents db tables based on ConfigType"""
 
-  def __init__(self, line=None, cmd=None, db_obj=None):
-    self.batchsize = 0
-    self.alpha = 1
-    self.beta = 0
-    self.forw = 1
-    self.back = 0
-    self.mode = 0
-    self.run = 0
-    self.in_d = 1
-    self.in_h = 1
-    self.in_w = 1
-    self.in_channels = 1
-    self.in_layout = 'NCHW'
-    self.num_dims = None
-    self.direction = None
-    self.save = 0
-    self.verify = 1
-    self._cmd = 'bnorm'
+  def __init__(self,
+               line: str = str(),
+               cmd: str = str(),
+               db_obj: BNConfig = None) -> None:
+    self.batchsize: int = 0
+    self.alpha: int = 1
+    self.beta: int = 0
+    self.forw: int = 1
+    self.back: int = 0
+    self.mode: int = 0
+    self.run: int = 0
+    self.in_d: int = 1
+    self.in_h: int = 1
+    self.in_w: int = 1
+    self.in_channels: int = 1
+    self.in_layout: str = 'NCHW'
+    self.num_dims: Optional[str] = None
+    self.direction: Union[str, int, None] = None
```

Comment:
This won't be optional, they will also be a singular type. Consult the defaults.